be/src/olap/olap_engine.cpp

```diff
@@ -835,6 +840,10 @@ OLAPStatus OLAPEngine::get_tables_by_id(
                 it = table_list->erase(it);
                 continue;
             }
+        } else if ((*it)->is_used()) {
```

Comment:
why not check if `(*it)->is_bad()`

I'm afraid when tablet is not bad and it is not used, you erase it. which is not what we want

Reply:
For now, `tablet is not used` means the store which this tablet on is bad(IO error). So I think using `(*it)->is_used()` is OK